Adapters/AdapterInterface.py

```diff
@@ -0,0 +1,174 @@
+            and callable(subclass.get_entityless_attribute)
+            and hasattr(subclass, "get_vo_attributes")
+            and callable(subclass.get_vo_attributes)
+            and hasattr(subclass, "get_vo_attributes_values")
+            and callable(subclass.get_vo_attributes_values)
+            and hasattr(subclass, "get_facility_attribute")
+            and callable(subclass.get_facility_attribute)
+            and hasattr(subclass, "get_facility_by_entity_id")
+            and callable(subclass.get_facility_by_entity_id)
+            and hasattr(subclass, "get_facility_by_client_id")
+            and callable(subclass.get_facility_by_client_id)
+            and hasattr(subclass, "get_users_groups_on_facility")
+            and callable(subclass.get_users_groups_on_facility)
+            and hasattr(subclass, "search_facilities_by_attribute_value")
+            and callable(subclass.search_facilities_by_attribute_value)
+            and hasattr(subclass, "get_facility_attributes")
+            and callable(subclass.get_facility_attributes)
+            and hasattr(subclass, "get_facility_attributes_values")
+            and callable(subclass.get_facility_attributes_values)
+            and hasattr(subclass, "get_user_ext_source")
+            and callable(subclass.get_user_ext_source)
+            and hasattr(subclass, "update_user_ext_source_last_access")
+            and callable(subclass.update_user_ext_source_last_access)
+            and hasattr(subclass, "get_user_ext_source_attributes")
+            and callable(subclass.get_user_ext_source_attributes)
+            and hasattr(subclass, "set_user_ext_source_attributes")
+            and callable(subclass.set_user_ext_source_attributes)
+            and hasattr(subclass, "get_member_status_by_user_and_vo")
+            and callable(subclass.get_member_status_by_user_and_vo)
+            and hasattr(subclass, "is_user_in_vo")
+            and callable(subclass.is_user_in_vo)
+            and hasattr(subclass, "get_resource_capabilities")
+            and callable(subclass.get_resource_capabilities)
+            and hasattr(subclass, "get_facility_capabilities")
+            and callable(subclass.get_resource_capabilities)
+            and hasattr(subclass, "remove_duplicate_entities")
+            and callable(subclass.remove_duplicate_entities)
+            or NotImplemented
+        )
+
+    @abc.abstractmethod
+    def get_perun_user(self, idp_entity_id: int, uids: List[int]) -> User:
+        """Get Perun user with at least one of the uids"""
+        raise NotImplementedError
+
+    def get_group_by_name(self, vo: Vo, name: str) -> Group:
+        """Get Group based on its name"""
+        raise NotImplementedError
+
+    def get_vo_by_short_name(self, vo_short_name: str) -> Vo:
+        """Get VO by its short name"""
+        raise NotImplementedError
+
+    def get_vo_by_id(self, id: int) -> Vo:
+        """Get VO by its id"""
+        raise NotImplementedError
+
```

Comment:
A more Pythonic approach might be:
```suggestion
    def get_vo(self, short_name=None, id=None) -> Vo:
        """Get VO"""
        raise NotImplementedError
```
which might be implemented like this:
```
    def get_vo(self, short_name=None, id=None) -> Vo:
        if id and short_name:
            raise IllegalArgumentError
        if id:
            # find by ID
        elif short_name:
            # find by short name
        else:
            raise MissingArgumentError
```